### PR TITLE
Adding cloudsnap delete event to longevity

### PR DESCRIPTION
**Pull request by Santhoshairody**
Add clouldsnap delete event to longevity
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** (optional)
Closes #
PTX-23570

**Special notes for your reviewer**:
http://jen-endurance.pwx.dev.purestorage.com/job/Users/job/santhoshm/job/IOProfile-update/66/console

### CODE REVIEW

File: drivers/volume/portworx/portworx.go [hunk 1]
```diff
@@ -1815,6 +1815,21 @@ func (d *portworx) GetCloudsnaps(volumeName string, params map[string]string) ([
...
 
+// GetCloudsnaps returns all the cloud snaps of the given volume
+func (d *portworx) GetCloudsnapsOfGivenVolume(volumeName string, sourceVolumeID string, params map[string]string) ([]*api.SdkCloudBackupInfo, error) {
+	var token string
+	token = d.getTokenForVolume(volumeName, params)
+	if val, hasKey := params[refreshEndpointParam]; hasKey {
+		refreshEndpoint, _ := strconv.ParseBool(val)
+		d.refreshEndpoint = refreshEndpoint
+	}
+	cloudSnapResponse, err := d.csbackupManager.EnumerateWithFilters(d.getContextWithToken(context.Background(), token), &api.SdkCloudBackupEnumerateWithFiltersRequest{SrcVolumeId: sourceVolumeID})
```
Comment (Rohit-PX) [inline drivers/volume/portworx/portworx.go:1826]: Just curious as to whether there is a wrapper function around this in `sched-ops`?

File: tests/testTriggers.go [hunk 2]
```diff
@@ -3500,6 +3504,82 @@ func TriggerCloudSnapShot(contexts *[]*scheduler.Context, recordChan *chan *Even
...
+	}()
+	setMetrics(*event)
+	stepLog := "Delete all cloudsnaps"
+	Step(stepLog, func() {
+		log.Infof(stepLog)
+        var volCloudsnapMap = make(map[string][]string)
+		for _, ctx := range *contexts {
+			vols, err := Inst().S.GetVolumeParameters(ctx)
+			log.Infof("Validating context: %v", ctx.App.Key)
+			log.Infof("Volumes : %v", vols)
```
Comment (Rohit-PX) [inline tests/testTriggers.go:3532]: Remove both these lines before merging.

File: tests/testTriggers.go [hunk 3]
```diff
@@ -3500,6 +3504,82 @@ func TriggerCloudSnapShot(contexts *[]*scheduler.Context, recordChan *chan *Even
...
+			UpdateOutcome(event, err)
+			for vol, params := range vols {
+				log.Infof("Volume Name : %s", vol)
```
Comment (Rohit-PX) [inline tests/testTriggers.go:3535]: Remove.

File: tests/testTriggers.go [hunk 4]
```diff
@@ -3500,6 +3504,82 @@ func TriggerCloudSnapShot(contexts *[]*scheduler.Context, recordChan *chan *Even
...
+				var cloudsnapIds []string
+				for _, bk := range csBksps {
+					log.Infof("Before Backup ID : %s having status : %v, Source volume: %s", bk.Id, bk.Status, bk.SrcVolumeName)
+					log.Infof("bk.GetMetadata() : %v , Source volume: %s", bk.GetMetadata(), bk.SrcVolumeName)
+					cloudsnapIds = append(cloudsnapIds, bk.Id)
+				}
+				if len(csBksps) > 0 {
+					volCloudsnapMap[vol] = cloudsnapIds
+					err = Inst().V.DeleteAllCloudsnaps(vol,csBksps[0].SrcVolumeId, params)
+					if err != nil && strings.Contains(err.Error(), "Key already exists") {
```
Comment (Rohit-PX) [inline tests/testTriggers.go:3549]:
```suggestion
					if err != nil && !strings.Contains(err.Error(), "Key already exists") {
					        UpdateOutcome(event, err)
					 }
```

File: tests/testTriggers.go [hunk 5]
```diff
@@ -3500,6 +3504,82 @@ func TriggerCloudSnapShot(contexts *[]*scheduler.Context, recordChan *chan *Even
...
+						log.Infof("KEY EXiST observed %v", err)
+						continue
```
Comment (Rohit-PX) [inline tests/testTriggers.go:3551]: Avoid continue by changing the condition above.

**Review by Rohit-PX — CHANGES_REQUESTED** [review]
Looks good otherwise.

File: drivers/volume/portworx/portworx.go (hunk 1 above)
Comment (Santhoshairody) [reply drivers/volume/portworx/portworx.go:1826]: Could not find any such method

**Review by Rohit-PX — APPROVED** [review]
LGTM.

**Pull request merged**